server/src/main/java/org/elasticsearch/common/compress/DeflateCompressor.java

```diff
@@ -129,4 +131,37 @@ public void close() throws IOException {
             }
         };
     }
+
+    private static final ThreadLocal<Inflater> inflaterRef = ThreadLocal.withInitial(() -> new Inflater(true));
```

Comment:
IMO it would be a good idea to add a comment about why these threadlocals are not used in the other methods of the class

Reply:
++